### ISSUE TITLE
pnpm 9 lockfile support

### ISSUE THREAD

**Issue by pawelblaszczyk5** [vercel/turbo#7993]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

https://github.com/pawelblaszczyk5/pnpm-prune-repro

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Which canary version will you have in your reproduction?

1.13.3-canary.1

### Describe the Bug

pnpm introduced a new lockfile format in the v9 major version. Formatting changes seems to mess up `turbo prune`, all stuff under `packages` key in output `pnpm-lock.yaml` is missing

### Expected Behavior

`turbo prune` should properly prune new pnpm lockfile format

### To Reproduce

- Update pnpm to v9
- Recreate lockfile so it utilizes the new `9.0` version
- Try to run `turbo prune some-package`
- Observe the prune output lockfile `packages` key being empty - it fails running `pnpm i --frozen-lockfile` because of that

### Additional context

I've pushed to the reproduction repo the lockfile after pruning before and after updating to pnpm v9. You can check the difference between `pnpm-lock-pruned-old.yaml` and `pnpm-lock-pruned-new.yaml`

**Comment by anthonyshew** [comment]
Thanks for the report! It's expected that this would fail, given it's a new format, after all. We're aware of it and have it on our roadmap.

**Comment by pawelblaszczyk5** [comment]
> Thanks for the report! It's expected that this would fail, given it's a new format, after all. We're aware of it and have it on our roadmap.

Yeah, I’m not surprised by this not working on day 1 of pnpm major release 😄 Just wanted to create this so it’s tracked somewhere and to make sure you’re aware of this. Thanks a lot!

**Comment by chris-olszewski** [comment]
This should be supported with `1.13.3-canary.2`.

If anyone encounters any issues with pnpm 9, please reopen this issue along with a reproduction repo.

**Comment by VojtechVidra** [comment]
Hello, I'm still having issus with pnpm 9. You can try by running `pnpm turbo prune --scope=web` in the following branch: https://github.com/RBND-studio/flows-cloud/pull/266

Hopefully it's not too complicated for you since it's not a minimal reproduction. Thanks!

**Comment by StuClift** [comment]
Using https://github.com/vercel/turbo/tree/main/examples/kitchen-sink
```console
$ cd examples/kitchen-sink
$ rm pnpm-lock.yaml
$ corepack use pnpm@9.0.2
Installing pnpm@9.0.2 in the project...

Scope: all 10 workspace projects
Packages: +21 -21
+++++++++++++++++++++---------------------
Progress: resolved 1274, reused 1187, downloaded 0, added 0, done
Done in 2.1s
$ pnpm update turbo@1.13.3-canary.2
Packages: +3 -3
+++---
Progress: resolved 1274, reused 1187, downloaded 0, added 0, done

devDependencies:
- turbo 1.12.4
+ turbo 1.13.3-canary.2

Done in 2.8s
$ pnpm exec turbo prune admin
Generating pruned monorepo for admin in /Users/stuart.clift/Development/turbo/examples/kitchen-sink/out
 - Added @repo/eslint-config
 - Added @repo/jest-presets
 - Added @repo/typescript-config
 - Added @repo/ui
 - Added admin
  × error parsing dependency path: error Eof at: )(eslint-import-resolver-node@0.3.9)(eslint-import-resolver-
  │ typescript@3.6.1(@typescript-eslint/parser@6.12.0(eslint@8.57.0)(typescript@5.3.3))(eslint-plugin-import@2.29.0)
  │ (eslint@8.57.0))(eslint@8.57.0)
  ╰─▶ error Eof at: )(eslint-import-resolver-node@0.3.9)(eslint-import-resolver-typescript@3.6.1(@typescript-eslint/
      parser@6.12.0(eslint@8.57.0)(typescript@5.3.3))(eslint-plugin-import@2.29.0)(eslint@8.57.0))(eslint@8.57.0)
$ head -1 pnpm-lock.yaml
lockfileVersion: '9.0'
$ pnpm -v
9.0.2
$ turbo --version
1.13.3-canary.2
```

**Comment by schowdhuri** [comment]
Can confirm, I see the same error as @StuClift 

```sh
Generating pruned monorepo for @app/api in /app/out
 - Added @app/api
  x error parsing dependency path: error Eof at: )(react@18.2.0)
  `-> error Eof at: )(react@18.2.0)
The command '/bin/sh -c turbo prune --scope=@app/api --docker' returned a non-zero code: 1
```

**Comment by chris-olszewski** [comment]
#8003 was released in[v1.13.3-canary.3](https://github.com/vercel/turbo/releases/tag/v1.13.3-canary.3) and should address issues with parsing dependencies with nested peer dependencies.

**Comment by jordanebelanger** [comment]
@chris-olszewski Any eta on how long this will be in a major, I am a bit hesitant to pin turbo or pnpm in a bunch of place in my project if this is coming soon.

**Comment by chris-olszewski** [comment]
@jordanebelanger 1.13.3 just got released this morning

**Comment by thexpand** [comment]
I'm still getting a warning when I build on Vercel, even using the latest turbo version in my turbo repo:
```
Detected `pnpm-lock.yaml` version 9 generated by pnpm 8
Running "install" command: `pnpm install`...
Scope: all 8 workspace projects

WARN  Ignoring not compatible lockfile at /vercel/path0/pnpm-lock.yaml
```

**Comment by rajatkapoor** [comment]
> I'm still getting a warning when I build on Vercel, even using the latest ([1.13.3](https://www.npmjs.com/package/turbo/v/1.13.3)) turbo version in my turbo repo:
> 
> ```
> Detected `pnpm-lock.yaml` version 9 generated by pnpm 8
> Running "install" command: `pnpm install`...
> Scope: all 8 workspace projects
> 
> WARN  Ignoring not compatible lockfile at /vercel/path0/pnpm-lock.yaml
> ```

@thexpand This workaround  helped me

Setting ENABLE_EXPERIMENTAL_COREPACK=1 in the env helps me get around this issue if you have set 
```
  "packageManager": "pnpm@9.0.6"
``` 
in your package.json

You can follow the steps as listed here: https://vercel.com/docs/deployments/configure-a-build#corepack

**Comment by pawelblaszczyk5** [comment]
I verified and it seems to work correctly with latest turbo version, closing the issue, tyvm!

**Comment by yannickschuchmann** [comment]
Just as a heads up. I came here because I also had that warning `WARN  Ignoring not compatible lockfile at /vercel/path0/pnpm-lock.yaml`, so I set the `packageManager` and `ENABLE_EXPERIMENTAL_COREPACK=1`. 

But it was complaining that the `packageManager` would still not be set in `package.json`. 
**Now the interesting part**: turned out since I've changed the root directory to a subfolder that it expected to find the `packageManager` field in a `package.json` still on root-root `.` level instead of `./subfolder`. 
Only when I added an empty `package.json` to my monorepo's root level with the `packageManager` set it worked.

Sorry if this is difficult to read. It's quite late here. Thing is tho it means that using multiple `packageManager` values / version in a monorepo is not possible with the current situation.
